src/lib/options/validate-options.ts

```diff
@@ -0,0 +1,170 @@
+import type { Options } from './index';
+
+const boolOpts = [
+  'makeCumulative',
+  'loop',
+  'showIcons',
+  'showGroups',
+  'mouseControls',
+  'keyboardControls',
+  'autorun',
+  'injectStyles',
+  'highlightBars',
+  'selectBars',
+  'fixedScale',
+] as const;
+const numberOpts = [
+  'labelsWidth',
+  'tickDuration',
+  'topN',
+  'minHeight',
+  'minWidth',
+  'marginTop',
+  'marginRight',
+  'marginBottom',
+  'marginLeft',
+] as const;
+const strOpts = ['theme', 'startDate', 'endDate'] as const;
+const strOrNumberOpts = ['colorSeed', 'inputHeight', 'inputWidth', 'height', 'width'] as const;
+const strOrFuncOpts = ['title', 'subTitle', 'dateCounter', 'caption'] as const;
+
+const validDataShapes = ['long', 'wide'];
+const validDataTypes = ['json', 'csv', 'tsv', 'xml'];
+const validLabelsPositions = ['inside', 'outside'];
+const validControlButtons = ['all', 'play', 'none'];
+const validOverlays = ['all', 'play', 'repeat', 'none'];
+const validFillDateGapsIntervals = [null, 'year', 'month', 'day'];
+const validFillDateGapsValues = ['last', 'interpolate'];
+
+export function validateOptions(options: Partial<Options>): Partial<Options> {
+  const newOptions: Partial<Options> = {};
+
+  // Validate boolean options
+  boolOpts.forEach((opt) => {
```

Comment:
we may want to allow string representation of booleans (e.g. in case they are supplied as strings from a react prop `<RacingBars autorun="false" />`).

```ts
// Validate boolean options
boolOpts.forEach((opt) => {
  if (!is(options[opt], 'boolean')) return;
  newOptions[opt] = options[opt] === 'false' || options[opt] === false ? false : true;
});
```

I may be wrong about this!

Reply:
To handle this case, I updated Options type

```
type BooleanLike = boolean | 'true' | 'false';

export interface Options {
  . . .
  makeCumulative: BooleanLike;
  loop: BooleanLike;
  showIcons: BooleanLike;
  showGroups: BooleanLike;
  mouseControls: BooleanLike;
  keyboardControls: BooleanLike;
  autorun: BooleanLike;
  injectStyles: BooleanLike;
  fixedScale: BooleanLike;
  highlightBars: BooleanLike;
  selectBars: BooleanLike;
  . . .
}
```

Reply:
@hatemhosny Due to these changes, i need to make changes in [prepare-data.ts](https://github.com/AhmedElbohoty/racing-bars/blob/735e4bbcd429a309e6a7052fcb7464e243396871/src/lib/worker/prepare-data.ts#L21). Kindly check it.

Reply:
> To handle this case, I updated Options type
> 
> ```
> type BooleanLike = boolean | 'true' | 'false';
> 
> export interface Options {
>   . . .
>   makeCumulative: BooleanLike;
>   loop: BooleanLike;
>   showIcons: BooleanLike;
>   showGroups: BooleanLike;
>   mouseControls: BooleanLike;
>   keyboardControls: BooleanLike;
>   autorun: BooleanLike;
>   injectStyles: BooleanLike;
>   fixedScale: BooleanLike;
>   highlightBars: BooleanLike;
>   selectBars: BooleanLike;
>   . . .
> }
> ```

I don't think we should do this.
This is a part of the public API. It would be confusing to enforce such a type. I just wanted to handle the case if the user passes a string that can be parsed as boolean. I did not mean that we recommend or enforce this.

Reply:
I believe the only way to accommodate this behavior right now is to ignore TypeScript on this line, which I’m not fond of. However, we might need to proceed this way until the options type is updated in the future.